src/content/docs/it/core-concepts/astro-components.mdx

```diff
@@ -0,0 +1,347 @@
+
+Puoi utilizzare lo script del componente per scrivere qualsiasi codice JavaScript necessario per eseguire il rendering del tuo modello. Ciò può includere:
+
+- importare altri componenti Astro
+- importando altri componenti del framework, come React
+- importazione di dati, come un file JSON
+- recupero di contenuti da un'API o da un database
+- creare variabili a cui farai riferimento nel tuo modello
+
+
+```astro title="src/components/MyComponent.astro"
+---
+import SomeAstroComponent from '../components/SomeAstroComponent.astro';
+import SomeReactComponent from '../components/SomeReactComponent.jsx';
+import someData from '../data/pokemon.json';
+
+// Accede alle proprietà del componente passato, come `<X title="Hello, World" />`
+const {title} = Astro.props;
+// Recupera dati esterni, anche da un'API o un database privato
+const data = await fetch('SOME_SECRET_API_URL/users').then(r => r.json());
+---
+<!-- Il tuo modello qui! -->
+```
+
+Il separatore del codice è progettato per garantire che il codice JavaScript che scrivi al suo interno sia "recintato". Non sfuggirà alla tua applicazione frontend né cadrà nelle mani dell'utente. Puoi scrivere qui in sicurezza codice costoso o sensibile (come una chiamata al tuo database privato) senza preoccuparti che finisca mai nel browser dell'utente.
+
+:::tip
+Puoi persino scrivere TypeScript nello script del tuo componente!
+:::
+
+### Il modello del componente
+
+Il modello del componente si trova sotto il separatore del codice e determina l'output HTML del componente.
+
+Se scrivi HTML semplice qui, il tuo componente rendererà quell'HTML in qualsiasi pagina Astro in cui viene importata e utilizzata.
+
+Tuttavia, [la sintassi del modello del componente Astro](/it/core-concepts/astro-syntax/) supporta anche le **espressioni JavaScript**, Astro [`<style>`](/it/guides/styling/#styling-in -astro) e tag [`<script>`](/it/guides/client-side-scripts/#using-script-in-astro), **componenti importati** e [**direttive speciali Astro** ](/it/reference/directives-reference/). I dati e i valori definiti nello script del componente possono essere utilizzati nel modello del componente per produrre HTML creato dinamicamente.
+
+```astro title="src/components/MyFavoritePokemon.astro"
+---
+// Il tuo script del componente qui!
+import Banner from '../components/Banner.astro';
+import ReactPokemonComponent from '../components/ReactPokemonComponent.jsx';
+const myFavoritePokemon = [/* ... */];
+const { title } = Astro.props;
+---
+<!-- Commenti HTML supportati! -->
+{/*Anche la sintassi dei commenti JS è valida! */}
+
+<Banner />
+<h1>Hello, world!</h1>
+
+<!-- Utilizza proprietà e altre variabili dallo script del componente: -->
+<p>{title}</p>
+
+<!-- Includi altri componenti del framework dell'interfaccia utente con una direttiva `client:` per idratare: -->
```

Comment:
```suggestion
<!-- Includi componenti di altri framework UI con una direttiva `client:` per idratare: -->
```

See my comment at line 347 in 
`src/content/docs/it/core-concepts/astro-components.mdx`